Index: django/template/__init__.py
===================================================================
--- django/template/__init__.py	(revision: 13865)
+++ django/template/__init__.py	(working copy)
@@ -240,6 +240,10 @@
                 token = Token(TOKEN_COMMENT, '')
         else:
             token = Token(TOKEN_TEXT, token_string)
+        # token.source is a tuple, this contains origin object and
+        # the range of the columns where is this token.
+        # If TEMPLATE_DEBUG = False we don't need it, therefore we set with (-1, -1)
+        token.source = self.origin, (-1, -1)
         return token
 
 class Parser(object):
@@ -311,6 +315,7 @@
                 raise TemplateSyntaxError("%r must be the first tag in the template." % node)
         if isinstance(nodelist, NodeList) and not isinstance(node, TextNode):
             nodelist.contains_nontext = True
+        node.source = token.source
         nodelist.append(node)
 
     def enter_command(self, command, token):
Index: django/template/loader.py
===================================================================
--- django/template/loader.py	(revision: 13865)
+++ django/template/loader.py	(working copy)
@@ -79,10 +79,9 @@
         return self.loader(self.loadname, self.dirs)[0]
 
 def make_origin(display_name, loader, name, dirs):
-    if settings.TEMPLATE_DEBUG and display_name:
+    if display_name:
         return LoaderOrigin(display_name, loader, name, dirs)
-    else:
-        return None
+    return None
 
 def find_template_loader(loader):
     if isinstance(loader, (tuple, list)):

